File: Node.java
package Folder;

import java.util.*;

public class Node {
    
    private int row;
    private int col;
    public List<Edge> neighbors;

    public Node(int row, int col) {
        super();
        this.row = row;
        this.col = col;
        neighbors= new ArrayList<>();
    }
////////////////////////////////////////////////////////////////////////////
        public class Edge{
            public int g;
            public double f;
            public double h;
            public Node node;

        Edge(Node node , int g){
            this.node = node;
            this.g=g;
            this.h=calculate_Manhattan_Heuristic();
            setF();
         }
      public double getH() {
        return h;
        }

        public void setH(Double h) {
        this.h = h;
        }

        public int getG() {
        return g;
        }

        public double getF() {
        return f;
        }

        public void setF() {
        this.f = getG()+getH();
        }
         public double calculate_Manhattan_Heuristic() {
            this.h = Math.abs(node.getRow() - getRow()) + Math.abs(node.getCol() - getCol());
            return this.h;
        }
        //TODO Add 2 other heu

        @Override
        public String toString() {
            return "Edge{" + "F=" + getF() + ", G=" + getG() + " H= "+getH();
        }
    }
////////////////////////////////////////////////////////////////////////////

    public void addBranch(int weight, Node node){
        Edge newEdge = new Edge(node , weight);
        neighbors.add(newEdge);
  }
    

    @Override
    public String toString() {
        return "Node{" + "row=" + row + ", col=" + col + " neighbors: "+neighbors;
    }
    public int getRow() {
        return row;
    }

    public void setRow(int row) {
        this.row = row;
    }

    public int getCol() {
        return col;
    }

    public void setCol(int col) {
        this.col = col;
    }
    public List<Edge> getNeighbor()
    {
        return neighbors;
    }
}

File: Node.py
import math
class Node: def __init__(self, row, col): self.row = row
self.col = col
self.neighbors = []
# //////////////////////////////////////////////////////////////////////////
class Edge: DigonalCost = 5
EculedianCost = 1
def __init__(self, node, g): self.node = node
self.g = g
self.setDf(self.Dh = self.calculate_Digonal_Heuristic())
self.setEf(self.Eh = self.calculate_Euclidean_Heuristic())
self.setMf(self.Mh = self.calculate_Manhattan_Heuristic())
def setMf(self, Mh): self.Mf = Mh + self.g
def setEf(self, Eh): self.Ef = self.Eh + self.g
def setDf(self, Dh): self.Df = self.Dh + self.g
def calculate_Manhattan_Heuristic(self): self.Mh = abs(self.node.getRow() - self.getRow()) + abs(self.node.getCol() - self.getCol())
return self.Mh
def calculate_Euclidean_Heuristic(self): x = math.pow(abs(self.getRow() - self.node.getRow()), 2)
y = math.pow(abs(self.getCol() - self.node.getCol()), 2)
self.Eh = self.EculedianCost * mathsqrt(x + y)
return self.Eh
def calculate_Digonal_Heuristic(self): dx = abs(self.getRow() - self.node.getRow())
dy = abs(self.getCol() - self.node.getCol())
self.Eh = self.DigonalCost * max(dx, dy)
return self.Eh
# //////////////////////////////////////////////////////////////////////////
def addBranch(self, weight, node): newEdge = Edge(node, weight)
self.neighbors.add(newEdge)
def toString(self): return "Node{" + "row=" + str(self.row) + ", col=" + str(self.col) + " neighbors: " + str(self.neighbors)
def getRow(self): return self.row
def setRow(self, row): self.row = row
def getCol(self): return self.col
def setCol(self, col): self.col = col
def getNeighbor(self): return self.neighbors